compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -720,7 +719,7 @@ readDalfFromFile dalfFile = do
             protoPkg <- case Proto.fromByteString bytes of
                 Left err -> fail (show err)
                 Right a -> pure a
-            case decodeScenarioModule (TL.pack $ LF.renderMinorVersion $ LF.versionMinor lfVersion) protoPkg of
+            case decodeScenarioModule lfVersion protoPkg of
```

Comment:
this is still the `decodeScenarioModule` from `...DecodeV1`, right? i think this codepath is only reached with `daml build --incremental`, so this will fail on an LFv2 project. The flag is exposed by `daml build --help`, so we can't argue it's just internal, but it looks like we don't mention it in [docs.daml.com](https://docs.daml.com), and there are only a handful of tests for it in https://github.com/digital-asset/daml/blob/main/compiler/damlc/tests/src/DA/Test/IncrementalBuilds.hs. 

That is to say, I think we should have a `decodeScenarioModule` that works with any major version, but perhaps we should also reconsider the `--incremental` flag given that it doesn't appear to be widely used.

Reply:
:facepalm: and you even told me about it last week! Done.